bundles/org.eclipse.wst.xml.xpath2.processor/src/org/eclipse/wst/xml/xpath2/processor/internal/XPathFlex.java

```diff
@@ -1317,7 +1318,7 @@ else if (zzAtEOF) {
           }
         case 170: break;
         case 49: 
-          { return symbol(XpathSym.DOUBLE, new Double(yytext()));
+          { return symbol(XpathSym.DOUBLE, LiteralUtils.parseDouble(yytext()));
```

Comment:
I believe you are putting the modification on the wrong file here.  Should be in https://github.com/sharwell/webtools.sourceediting.xpath/blob/75f20765480c1704043ef68dc59d0d5db5bf603b/bundles/org.eclipse.wst.xml.xpath2.processor/grammars/xpath.lex#L180